source/dol/bases/d_ac_py_key.cpp

```diff
@@ -0,0 +1,355 @@
+    return mTriggeredButtons & BUTTON_A;
+}
+
+int dAcPyKey_c::buttonCross() const {
+    return mDownButtons & BUTTON_CROSS;
+}
+
+int dAcPyKey_c::triggerCross() const {
+    return mTriggeredButtons & BUTTON_CROSS;
+}
+
+int dAcPyKey_c::buttonUp() const {
+    return mDownButtons & BUTTON_UP;
+}
+
+int dAcPyKey_c::buttonDown() const {
+    return mDownButtons & BUTTON_DOWN;
+}
+
+int dAcPyKey_c::buttonLeft() const {
+    return mDownButtons & BUTTON_LEFT;
+}
+
+int dAcPyKey_c::buttonRight() const {
+    return mDownButtons & BUTTON_RIGHT;
+}
+
+int dAcPyKey_c::triggerUp() const {
+    return mTriggeredButtons & BUTTON_UP;
+}
+
+int dAcPyKey_c::triggerDown() const {
+    return mTriggeredButtons & BUTTON_DOWN;
+}
+
+int dAcPyKey_c::triggerLeft() const {
+    return mTriggeredButtons & BUTTON_LEFT;
+}
+
+int dAcPyKey_c::triggerRight() const {
+    return mTriggeredButtons & BUTTON_RIGHT;
+}
+
+int dAcPyKey_c::triggerTwo() const {
+    return mTriggeredButtons & BUTTON_TWO;
+}
+
+int dAcPyKey_c::buttonTwo() const {
+    return mDownButtons & BUTTON_TWO;
+}
+
+int dAcPyKey_c::triggerOne() const {
+    if (dGameKey_c::m_instance->keys[mRemoconID]->mAttachedExtension == 1) {
+        return mTriggeredButtons & BUTTON_B;
+    } else {
+        return mTriggeredButtons & BUTTON_ONE;
+    }
+}
+
+int dAcPyKey_c::buttonOne() const {
+    if (dGameKey_c::m_instance->keys[mRemoconID]->mAttachedExtension == 1) {
+        return mDownButtons & BUTTON_B;
+    } else {
+        return mDownButtons & BUTTON_ONE;
+    }
+}
+
+int dAcPyKey_c::buttonDush() const {
+    return mDownButtons & BUTTON_ONE;
+}
+
+int dAcPyKey_c::triggerAttack() const {
+    return triggerOne();
+}
+
+bool dAcPyKey_c::triggerEat() {
+    if (triggerOne()) {
+        mActionTriggered = true;
+        return true;
+    }
+    return false;
+}
+
+int dAcPyKey_c::buttonCarry() const {
+    return buttonOne();
+}
+
+bool dAcPyKey_c::triggerFire() {
+    if (triggerOne()) {
+        mActionTriggered = true;
+        return true;
+    }
+    return false;
+}
+
+int dAcPyKey_c::triggerJumpBase() const {
+    return triggerTwo();
+}
+
+bool dAcPyKey_c::triggerJump() {
+    if (triggerJumpBase()) {
+        mActionTriggered = true;
+        return true;
+    }
+    return false;
+}
+
+int dAcPyKey_c::buttonJump() {
+    if (mStatus & FLAG_FORCE_JUMP) {
+        return true;
+    }
+    if (mStatus & FLAG_FORCE_NO_JUMP) {
+        return false;
+    }
+    return buttonTwo();
+}
+
+int dAcPyKey_c::buttonYoshiJump() const {
+    return buttonTwo();
+}
+
+u8 dAcPyKey_c::triggerShakeJump() const {
+    return mTriggerShakeJump;
+}
+
+void dAcPyKey_c::clearShakeJump() {
+    mTriggerShakeJump = 0;
+    mActiveShakeJump = 0;
+}
+
+bool dAcPyKey_c::buttonWalk(int *direction) const {
+    if (buttonRight()) {
+        if (direction != nullptr) {
+            *direction = 0;
+        }
+        return true;
+    } else if (buttonLeft()) {
+        if (direction != nullptr) {
+            *direction = 1;
```

Comment:
Probably not the right PR to ask for this, but i believe at some point we should add an enum for directions, since these values are used very often in the game and people can easily confuse them.